### PR TITLE
Add pagination to `GET Package` endpoint

### LINKED ISSUE #91: Add pagination for `GET /package`
# Problem
Currently, this endpoint simply returns all releases matching the query criteria. This is nice and simple, but struggles for queries with larger result sets.
# Solution
Pagination! We have a few choices for how we implement this.
- https://pypi.org/project/fastapi-pagination/ we can try this. I've never used or heard about it.
- We can implement our own pagination. We would need to decide what kid of pagination we want (offset vs cursor).

### PR DESCRIPTION
**Pull request by BruDriguezz**
Closes #91

### CODE REVIEW

File: src/mainframe/endpoints/package.py [hunk 1]
```diff
@@ -173,11 +177,11 @@ def lookup_package_info(
         query = query.where(Scan.finished_at >= dt.datetime.fromtimestamp(since, tz=dt.timezone.utc))
 
     with session, session.begin():
-        data = session.scalars(query).unique().all()
-        packages = [Package.from_db(result) for result in data]
-        log.info("Package information queried")
-
-    return packages
+        log.info("Package info queried")
```
Comment (jonathan-d-zhang) [inline src/mainframe/endpoints/package.py:180]:
```suggestion
```

**Review by jonathan-d-zhang — CHANGES_REQUESTED** [review]
This change will cause the response model of the endpoint to change, right? have you made sure that bot will not break because of this?

**Review by jonathan-d-zhang — APPROVED** [review]
lgtm

**Review by jonathan-d-zhang — DISMISSED** [review]

**Review by jonathan-d-zhang — DISMISSED** [review]

File: src/mainframe/endpoints/package.py [hunk 2]
```diff
@@ -169,11 +173,13 @@ def lookup_package_info(
         query = query.where(Scan.finished_at >= dt.datetime.fromtimestamp(since, tz=dt.timezone.utc))
 
     with session, session.begin():
+        if page is not None and size is not None:
```
Comment (Robin5605) [inline src/mainframe/endpoints/package.py:176]:
```suggestion
        if page and size:
```

**Review by Robin5605 — APPROVED** [review]
pretty much exactly what i was going for, thanks!

**Review by import-pandas-as-numpy — APPROVED** [review]

**Pull request merged**